websocket/connection.go

```diff
@@ -78,38 +76,44 @@ type ManagedConnection struct {
 // that can only send messages to the endpoint it connects to.
 // The connection will continuously be kept alive and reconnected
 // in case of a loss of connectivity.
-func NewDurableSendingConnection(target string) *ManagedConnection {
-	return NewDurableConnection(target, nil)
+func NewDurableSendingConnection(target string, logger *zap.SugaredLogger) *ManagedConnection {
+	return NewDurableConnection(target, nil, logger)
 }
 
 // NewDurableConnection creates a new websocket connection, that
 // passes incoming messages to the given message channel. It can also
 // send messages to the endpoint it connects to.
 // The connection will continuously be kept alive and reconnected
 // in case of a loss of connectivity.
-func NewDurableConnection(target string, messageChan chan []byte) *ManagedConnection {
-	c := newConnection(target, messageChan)
+func NewDurableConnection(target string, messageChan chan []byte, logger *zap.SugaredLogger) *ManagedConnection {
+	websocketConnectionFactory := func(target string) (rawConnection, error) {
```

Comment:
Interestingly enough `target` is redundant here, since its immutable and can be bound here (thought it can be kept for expressiveness).